### PR TITLE
Update Dataflow internal client test

### PR DESCRIPTION
**Pull request by jrmccluskey**
Fixes a breakage causing the coverage precommit to go red. Updates the test to a later python version to exercise the code

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py [hunk 1]
```diff
@@ -998,12 +998,12 @@ def test_interpreter_version_check_passes_with_experiment(self):
 
   @mock.patch(
       'apache_beam.runners.dataflow.internal.apiclient.sys.version_info',
-      (3, 8, 2))
```
Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py:1001]: there is benefit to keep this aligned with the lowest supported version.

Hence question - do we drop Py3.8 support in the upcoming release or the following release? I forgot where we finished that conversation.

Comment (jrmccluskey) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py:1001]: No one really felt strongly either way, which is why this kept moving forward with the testing changes.

I can swing this back to 3.9 no problem

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py:1001]: reading https://github.com/apache/beam/blob/e00559140df6cb45989bc62356d3b512bd32a537/sdks/python/apache_beam/runners/dataflow/internal/apiclient.py#L85 we drop support in Dataflow for python 3.9 in the upcoming release.

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py:1001]: so if the plan was to "move tests to 3.9 now, but still support Python 3.8 for one more release", then we strayed from that plan

Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py:1001]: this test essentially tests that Python 3.8 is still supported on Dataflow.

Comment (jrmccluskey) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py:1001]: If that's actually how we want to go about it then that's fine, but I'm of the opinion that if we're not testing things we're better off moving forward with removing support. I can put 3.8 back on that list for now if that's preferred and we can move this discussion to a later PR actually removing 3.8 support from beam proper

Comment (jrmccluskey) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py:1001]: this file was probably updated too early anyway IMO, best to leave it for the in-package removal

**Review by tvalentyn — APPROVED** [review]

File: sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py (hunk 1 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py]: > If that's actually how we want to go about it then that's fine, but I'm of the opinion that if we're not testing things we're better off moving forward with removing support. 

yeah, I am of the same opinion as you; my understanding was that this was a compromise b/w start the cleanup work now and also delay the deprecation for a little longer.

**Review by kennknowles — APPROVED** [review]

**Comment by kennknowles** [comment]
Looks like you also need to re-instate this "ignore" comment: https://github.com/apache/beam/commit/bbd09d60665433871b1e6be6b3e5d39431d57147#diff-b16eff1945003295256595d8a30c02515ba24aebb2ad7c8bc81ace56ebc3aabeL243

**Comment by jrmccluskey** [comment]
> bbd09d6#diff-b16eff1945003295256595d8a30c02515ba24aebb2ad7c8bc81ace56ebc3aabeL243

Fixed in #32475

**Pull request merged**